src/main/java/me/coley/recaf/command/impl/Remap.java

```diff
@@ -48,8 +54,30 @@ public Void call() throws Exception {
 		mappings.setClearDebugInfo(noDebug);
 		mappings.setCheckFieldHierarchy(lookup);
 		mappings.setCheckMethodHierarchy(lookup);
-		Map<String, byte[]> mapped = mappings.accept(getWorkspace().getPrimary());
-		// TODO: If the primary has a "META-INF/MANIFEST.MF" update the main class if renamed
+
+		JavaResource primary = getWorkspace().getPrimary();
+		Map<String, byte[]> mapped = mappings.accept(primary);
+
+		byte[] manifestBytes = primary.getFiles().get("META-INF/MANIFEST.MF");
+		if (manifestBytes != null) {
+			info("Found manifest file!");
+			Manifest manifest = new Manifest(new ByteArrayInputStream(manifestBytes));
+			Attributes attr = manifest.getMainAttributes();
+			if (!attr.isEmpty()) {
+				String mainClass = attr.getValue("Main-Class").replaceAll("\\.", "/");
+				if (mapped.containsKey(mainClass)) {
+					info("Found Main-Class attribute!");
+					attr.putValue("Main-Class", new ClassReader(mapped.get(mainClass))
+							.getClassName().replaceAll("/", "\\."));
+					ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+					manifest.write(outputStream);
+					primary.getFiles().put("META-INF/MANIFEST.MF", outputStream.toByteArray());
+					outputStream.close();
+					info("Remapped manifest!");
```

Comment:
Logging level can probably be dropped to debug. And I don't think they need to end with `!` either.